sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -36,8 +37,8 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
     new ParseException(errorClass = "_LEGACY_ERROR_TEMP_0001", ctx)
   }
 
-  def insertOverwriteDirectoryUnsupportedError(ctx: InsertIntoContext): Throwable = {
-    new ParseException(errorClass = "_LEGACY_ERROR_TEMP_0002", ctx)
+  def insertOverwriteDirectoryUnsupportedError(): Throwable = {
+    SparkException.internalError("INSERT OVERWRITE DIRECTORY is not supported.")
```

Comment:
Is it really an internal error. Can't an user run the command as by the rule?
```
    | INSERT OVERWRITE LOCAL? DIRECTORY (path=stringLit)? tableProvider (OPTIONS options=propertyList)? #insertOverwriteDir

```

Reply:
1.Although the above exception will be triggered when executing `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir ` in AstBuilder, our user is using the `SparkSqlAstBuilder` class, which extends AstBuilder and has overridden the `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir` methods and will not trigger the above logic.

- AstBuilder `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir`
https://github.com/apache/spark/blob/014dd357656106cc8dc0fcfb30cbe067a714916c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L368-L382

- SparkSqlAstBuilder `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir`

Reply:
Although the above exception will be triggered when executing `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir ` in AstBuilder, our user is using the `SparkSqlAstBuilder` class, which extends AstBuilder and has overridden the `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir` methods and will not trigger the above logic.

- AstBuilder `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir`
https://github.com/apache/spark/blob/014dd357656106cc8dc0fcfb30cbe067a714916c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L368-L382

- SparkSqlAstBuilder `visitInsertOverwriteDir` and `visitInsertOverwriteHiveDir`
https://github.com/apache/spark/blob/88f69d6f92860823b1a90bc162ebca2b7c8132fc/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L802-L834

https://github.com/apache/spark/blob/88f69d6f92860823b1a90bc162ebca2b7c8132fc/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala#L848-L866